src/main/java/ca/team2706/frc/robot/commands/drivebase/DriverAssistVision.java

```diff
@@ -14,189 +15,335 @@
+                    return;
+                }
+
+                // If distance to target is above a reasonable value, it is likely a temporary glitch in the
+                // vision measurement so return from the current execute() cycle but don't abort the command 
+                // so another reading can be taken on the next execute() call
+                if (distanceCameraToTarget_Camera > Config.VISION_DISTANCE_MAX.value()) {
+                    Log.d("DAV: Distance to target too high. Rereading vision data.");
+                    return;
+                }
+
+                // Turn off ring light since we now have the data from vision
+                Log.d("DAV: Turning ring light off");
+                RingLight.getInstance().disableLight();
+
+                // Send request to generate trajectory in a separate task to avoid execute() overruns
+                Log.d("DAV: Generating trajectory");
+                Runnable task = new Runnable() {
+                    public void run() {
+                        generateTrajectoryRobotToTarget(distanceCameraToTarget_Camera, angYawTargetWrtCameraLOSCWpos, target);
+                    }
+                };
+                new Thread(task).start();
+
+                generateTrajectoryRequestStageComplete = true;
+            }
+        }
 
-        trajGenerated = true;
+        // Stage 3: Command Robot to Follow Trajectory
+        if ((followTrajectory == null) && (trajectory != null)) {
+            Log.d("DAV: Commanding robot to follow trajectory");
+            followTrajectory = new FollowTrajectory(1.0, 100, trajectory);
+            followTrajectory.start();
+        }
     }
 
     @Override
     public boolean isFinished() {
-        return (trajGenerated || commandAborted);
+        return ( ((followTrajectory != null) && followTrajectory.isFinished()) || commandAborted);
     }
 
     @Override
     public void end() {
+        Log.d("DAV: Calling end()");
+        if (followTrajectory != null) {
+            if (followTrajectory.isRunning()) {
+                followTrajectory.cancel();
+            }
+            followTrajectory = null;
+        }
+
+        // Ring light should already be off, but just make sure in case there were early exits
+        // due to errors
+        RingLight.getInstance().disableLight();
+
+        // Put vision back into driver mode
+        driverEntry = chickenVisionTable.getEntry("Driver");
+        findTapeEntry = chickenVisionTable.getEntry("Tape");
+        findCargoEntry = chickenVisionTable.getEntry("Cargo");
+        driverEntry.setBoolean(true);
+        findCargoEntry.setBoolean(false);
+        findTapeEntry.setBoolean(false);
+
+        // Vision group has requested that tapeDetected entry be set to false here
+        tapeDetectedEntry = chickenVisionTable.getEntry("tapeDetected");
+        tapeDetectedEntry.setBoolean(false); 
+
         // Go back to disabled mode
         DriveBase.getInstance().setDisabledMode();
```

Comment:
This should also be removed since the command isn't actually driving

Reply:
Removed.